interledger/rafiki
# feat: dockerize integration tests

This pull request is for documentation purposes only.

It is a draft of dockerizing the integration tests. Just opening this to highlight a few things.

Can run tests in docker using the `./scripts/run-integration.sh` command.

Main changes are:
- adds dockerfile for `/test/integration`
- adds test-runner service to docker compose for our jest tests
- updates `run-integration.sh` script to get `test-runner`exit status so the process can fail according to the jest tests (important for ci) and displays logs

Note, there are also  some changes to import from `apollo/client/core` instead of `apollo/client`. This was to solve a problem I was running into with apollo/client trying to reference react when using the pnpm-lock.yaml but ultimately didnt work.

This roughs out a general method for running tests in docker. Several things should probably be improved including:
- might need to copy pnpm-lock.yaml in dockerfile (might diverge from local env in meaningful way? not sure)
- try to use other Dockerfile optimizations
- copy over base jest/tsconfig instead of inlining (to prevent divergence from local environement)
- probably support running either in docker or on host in `run-integration.sh`
